postgrest_py/exceptions.py

```diff
@@ -0,0 +1,32 @@
+from typing import Dict
+
+
+class APIError(Exception):
+    """
+    Base exception for all API errors.
+    """
+
+    _raw_error: Dict[str, str]
+    message: str
+    code: str
+    hint: str
+    details: str
+
+    def __init__(self, error: Dict[str, str]) -> None:
+        self._raw_error = error
+        self.message = error["message"]
+        self.code = error["code"]
+        self.hint = error["hint"]
+        self.details = error["details"]
+        super().__init__(str(self))
```

Comment:
Use `Exception.__init__(self, str(self))` to avoid possible future problems with multiple inheritances